colossus/src/main/scala/colossus/util/Parsing.scala

```diff
@@ -6,22 +6,54 @@ import core.DataBuffer
 
 import akka.util.{ByteString, ByteStringBuilder}
 
+import scala.util.Try
+
 sealed trait ParseStatus
 case object Incomplete extends ParseStatus
 case object Complete extends ParseStatus
 
 class ParseException(message: String) extends Exception(message)
 
 case class DataSize(value: Long) extends AnyVal {
-  def MB: DataSize = value * 1024 * 1024
-  def KB: DataSize = value * 1024
+  def MB: DataSize = DataSize(value * 1024 * 1024)
+  def KB: DataSize = DataSize(value * 1024)
   def bytes = value
 }
 
 object DataSize {
-  implicit def longToDataSize(l: Long): DataSize = DataSize(l)
+  //implicit def longToDataSize(l: Long): DataSize = DataSize(l)
+
+  implicit class IntToSize(val amount : Int) extends DataSizeConversions[Int]
+  implicit class LongToSize(val amount : Long) extends DataSizeConversions[Long]
+
+  trait DataSizeConversions[T] {
+    def amount : T
+
+    def bytes = B
+    def B = DataSize(s"$amount B")
+    def MB = DataSize(s"$amount MB")
```

Comment:
There's definitely some duplication of logic here.  I think it makes sense to just put all these methods in `DataSize` itself and get rid of the implicit classes.  Also I'm not a huge fan of these using string parsing in these methods since it's pretty inefficient.